packages/Adapters/STKMesh/example/mesh_deformation.cpp

```diff
@@ -0,0 +1,414 @@
+    stk::mesh::Selector& moving_selector,
+    stk::mesh::Selector& interior_selector,
+    Teuchos::Array<double>& displace_field )
+{
+    // Get the fixed boundary nodes.
+    stk::mesh::BucketVector fixed_part_buckets = 
+        fixed_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> fixed_part_nodes;
+    stk::mesh::get_selected_entities( 
+        fixed_selector, fixed_part_buckets, fixed_part_nodes );
+    int fixed_num_part_nodes = fixed_part_nodes.size();
+
+    // Get the moving boundary nodes.
+    stk::mesh::BucketVector moving_part_buckets = 
+        moving_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> moving_part_nodes;
+    stk::mesh::get_selected_entities( 
+        moving_selector, moving_part_buckets, moving_part_nodes );
+    int moving_num_part_nodes = moving_part_nodes.size();
+
+    // Get the number of boundary nodes.
+    int num_boundary_nodes = 
+        fixed_num_part_nodes + moving_num_part_nodes;
+
+    // Get the interior nodes.
+    stk::mesh::BucketVector interior_part_buckets = 
+        interior_selector.get_buckets( stk::topology::NODE_RANK );
+    std::vector<stk::mesh::Entity> interior_part_nodes;
+    stk::mesh::get_selected_entities( 
+        interior_selector, interior_part_buckets, interior_part_nodes );
+    int interior_num_part_nodes = interior_part_nodes.size();
+
+    // Get the coordinate field.
+    const stk::mesh::FieldBase* coord_field_base= 
+	bulk_data.mesh_meta_data().coordinate_field();
+    const stk::mesh::Field<double,stk::mesh::Cartesian3d>* coord_field =
+	dynamic_cast<const stk::mesh::Field<double,stk::mesh::Cartesian3d>* >(
+	    coord_field_base);
+    double* coords;
+
+    // Get the fixed boundary node coordinates.
+    int f = 0;
+    for ( int n = 0; n < fixed_num_part_nodes; ++n )
+    {
+        f = n;
+
+        coords = stk::mesh::field_data( *coord_field, fixed_part_nodes[n] );
+        coords[0] += displace_field[f*3];
+        coords[1] += displace_field[f*3+1];
+    }
+
+    // Get the moving boundary node coordinates.
+    for ( int n = 0; n < moving_num_part_nodes; ++n )
+    {
+        f = n + fixed_num_part_nodes;
+
+        coords = stk::mesh::field_data( *coord_field, moving_part_nodes[n] );
+        coords[0] += displace_field[f*3];
+        coords[1] += displace_field[f*3+1];
+    }
+    
+    // Get the interior node coordinates.
+    for ( int n = 0; n < interior_num_part_nodes; ++n )
+    {
+        f = n + num_boundary_nodes;
+
+        coords = stk::mesh::field_data( *coord_field, interior_part_nodes[n] );
+        coords[0] += displace_field[f*3];
+        coords[1] += displace_field[f*3+1];
+    }
+}
+
+//---------------------------------------------------------------------------//
+// Example driver.
+//
+// This example takes a mesh (mesh_deform_2d.exo) and deforms it based on the
+// movement of nodes on the interior boundary using spline interpolation. The
```

Comment:
if you take a look at the exodus file the mesh is a rectangle with a rectangle cut out of the middle. the boundary of the interior rectangle is the interior boundary. this is the boundary that moves.